setup.py

```diff
@@ -0,0 +1,67 @@
+"""
+Copyright 2018 SuperDARN
+
+setup.py
+2018-11-05
+To setup pyDARN as a third party library. Include installing need libraries for
+running the files.
+
+author:
+Marina Schmidt
+"""
+
+from distutils.core import setup
+from setuptools import setup, find_packages
+from os import path
+import sys
+from subprocess import check_call
+from setuptools.command.install import install
+
+# TODO: currently not implemented due to some challenges
+# with C API and memory leaks.
+#rstmodule = Extension('dmap',
+#                      sources= ['dmap.c'])
+
+class initialize_submodules(install):
+    def run(self):
```

Comment:
I think we need to try to reproduce what [`setuptools.command.install.run` does here](https://github.com/pypa/setuptools/blob/master/setuptools/command/install.py#L58-L67). This code basically does it, but I think it might need a couple changes. See other comments.